x-pack/plugin/security/qa/multi-cluster/src/javaRestTest/java/org/elasticsearch/xpack/remotecluster/RemoteClusterSecurityLicensingAndFeatureUsageRestIT.java

```diff
@@ -0,0 +1,223 @@
+
+            // Check that the feature is tracked on both QC and FC.
+            assertFeatureTracked(client());
+            assertFeatureTracked(fulfillingClusterClient);
+
+            final String licenseDowngradeTestCase = randomFrom("downgrade-fc-only", "downgrade-qc-only", "downgrade-both");
+            switch (licenseDowngradeTestCase) {
+                case "downgrade-fc-only" -> {
+                    // Downgrade fulfilling cluster license to BASIC and run CCS
+                    deleteLicenseFromCluster(fulfillingClusterClient);
+                    assertRequestFailsDueToUnsupportedLicense(() -> performRequestWithRemoteSearchUser(searchRequest));
+                }
+                case "downgrade-qc-only" -> {
+                    // Downgrade querying cluster license to BASIC and run CCS
+                    deleteLicenseFromCluster(client());
+                    assertRequestFailsDueToUnsupportedLicense(() -> performRequestWithRemoteSearchUser(searchRequest));
+                }
+                case "downgrade-both" -> {
+                    // Downgrade both fulfilling and querying cluster licenses to BASIC and run CCS
+                    deleteLicenseFromCluster(fulfillingClusterClient);
+                    assertRequestFailsDueToUnsupportedLicense(() -> performRequestWithRemoteSearchUser(searchRequest));
+
+                    deleteLicenseFromCluster(client());
+                    assertRequestFailsDueToUnsupportedLicense(() -> performRequestWithRemoteSearchUser(searchRequest));
+                }
+                default -> throw new IllegalStateException("Unexpected value: " + licenseDowngradeTestCase);
+            }
+        }
+    }
+
+    private void assertRequestFailsDueToUnsupportedLicense(ThrowingRunnable runnable) {
+        var exception = expectThrows(ResponseException.class, runnable);
+        assertThat(exception.getResponse().getStatusLine().getStatusCode(), equalTo(403));
+        assertThat(
+            exception.getMessage(),
+            containsString("current license is non-compliant for [" + SECURITY_CROSS_CLUSTER_ACCESS_FEATURE_NAME + "]")
+        );
+    }
+
+    private void startTrialLicense(RestClient client) throws IOException {
+        Request request = new Request("POST", "/_license/start_trial?acknowledge=true");
+        request.setOptions(RequestOptions.DEFAULT.toBuilder().addHeader("Authorization", basicAuthHeaderValue(USER, PASS)));
+        Response response = client.performRequest(request);
+        assertOK(response);
+    }
+
+    private void startBasicLicense(RestClient client) throws IOException {
+        Request request = new Request("POST", "/_license/start_basic?acknowledge=true");
+        request.setOptions(RequestOptions.DEFAULT.toBuilder().addHeader("Authorization", basicAuthHeaderValue(USER, PASS)));
+        Response response = client.performRequest(request);
+        assertOK(response);
+    }
```

Comment:
These methods are not used in current test cases.

Reply:
:facepalm: I missed to remove these. I was actually attempting to test something similar to the case which you described [in this comment](https://github.com/elastic/elasticsearch/pull/94638#discussion_r1150017258). I failed because my approach was inverted. I tried testing "allowed" -> "not allowed" -> "allowed again", which isn't possible.